server/syncSubs.ts

```diff
@@ -86,6 +88,59 @@ async function syncSubscribers() {
       process.exit(1);
     }
   }
+
+  const client = new DiscordBot({
+    intents: [Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MEMBERS],
+  });
+  try {
+    await client.login(config.DISCORD_BOT_TOKEN);
+    const guild = client.guilds.cache.get(config.DISCORD_SERVER_ID);
+    const members = await guild?.members.fetch();
+
+    // from discord API get username and discriminator of all users on discord server without a sub role
+    const membersWithoutRole = members
```

Comment:
What if we just joined the subscriber and discord links table and applied the role to all those users? Since the discord has thousands of members, might be more efficient? As mentioned before I don't think cleanup is essential and we can add a separate process for that later if needed.

Reply:
Yes, we could do that. But we will have to add a `hasRole` column to the discord table so we don't keep trying to assign a role to users who already have one over and over again.

Reply:
this may be moot if we just do the role add on link

Reply:
but to answer the question, can we check-then-assign for each user?

Reply:
Yes, we can check role status for each user with an individual discord API call, if this is what you're asking.